Fix from upstream. Keep icons constant when adding a stream to a playlist.
http://bugs.gentoo.org/show_bug.cgi?id=270564
http://developer.berlios.de/bugs/?func=detailbug&bug_id=15722&group_id=7323

diff -ur sonata-1.6.2.orig/sonata/artwork.py sonata-1.6.2/sonata/artwork.py
--- sonata-1.6.2.orig/sonata/artwork.py	2009-04-02 06:57:29.000000000 +0300
+++ sonata-1.6.2/sonata/artwork.py	2009-05-25 14:14:48.000000000 +0300
@@ -228,6 +228,8 @@
 		# Search through the rows in the library to see
 		# if we match the currently playing song:
 		play_artist, play_album = library_get_data(cache_key, 'artist', 'album')
+		if play_artist is None and play_album is None:
+			return
 		for row in self.lib_model:
 			artist, album, path = library_get_data(row[1], 'artist', 'album', 'path')
 			if unicode(play_artist).lower() == unicode(artist).lower() \
